core/src/main/java/kafka/server/share/ShareFetchUtils.java

```diff
@@ -117,4 +119,17 @@ static long offsetForEarliestTimestamp(TopicIdPartition topicIdPartition, Replic
                 Optional.empty(), true).timestampAndOffsetOpt();
         return timestampAndOffset.isEmpty() ? (long) 0 : timestampAndOffset.get().offset;
     }
+
+    /**
+     * The method is used to get the offset for the latest timestamp for the topic-partition.
+     *
+     * @return The offset for the latest timestamp.
+     */
+    static long offsetForLatestTimestamp(TopicIdPartition topicIdPartition, ReplicaManager replicaManager) {
+        // Isolation level is set to READ_UNCOMMITTED, matching with that used in share fetch requests
+        Option<FileRecords.TimestampAndOffset> timestampAndOffset = replicaManager.fetchOffsetForTimestamp(
+            topicIdPartition.topicPartition(), ListOffsetsRequest.LATEST_TIMESTAMP, new Some<>(IsolationLevel.READ_UNCOMMITTED),
+            Optional.empty(), true).timestampAndOffsetOpt();
+        return timestampAndOffset.get().offset;
```

Comment:
Do we need a check that `timestampAndOffset` has the object?

Reply:
Thanks for the review. As per the fetchOffsetForTimestamp method in Partition.scala, the method actually never returns a None when we try to fetch the latest offset, but I will still add the check just to be sure.